### PR TITLE
Api formatting

### PR DESCRIPTION
**Pull request by MarcosX**
Add documentation on standard data formats and general request/response modeling.

#1

### CODE REVIEW

File: api/formatting.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    "id" : 1
+    "startsOn" : "2017-10-16",
+    //...
+  },
+  {
+    "id" : 2
+    "startsOn" : "2017-10-16",
+    //...
+  },
+  //...
```
Comment (lucmir) [inline api/formatting.md]: This is an invalid JSON. Maybe we should replace it with something like:
```
// avoid this
[
  {
    "id" : 1
    "startsOn" : "2017-10-16",
    //...
  },
  {
    "id" : 2
    "startsOn" : "2017-10-16",
    //...
  },
  //...
]
```

File: api/formatting.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+}
```
Comment (lucmir) [inline api/formatting.md]: This is an invalid JSON. Maybe we should replace it with something like:
```
// avoid this
[
  {
    "id" : 1
    "startsOn" : "2017-10-16",
    //...
  },
  {
    "id" : 2
    "startsOn" : "2017-10-16",
    //...
  },
  //...
]
```

File: api/formatting.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+Example:
+
+```
+// avoid this
+{
+  "data" : {
+    "leave" : {
+    }
+  }
+}
```
Comment (lucmir) [inline api/formatting.md]: Just a little suggestion: include a blank line between the "avoid this" and "do this instead" examples.

File: api/formatting.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    "startsOn" : "2017-10-16",
+    //...
+  },
+  //...
+}
+// do this instead
+{
+  "leave" : [
+    {
+      "id" : 1
```
Comment (lucmir) [inline api/formatting.md]:
```
"id": 1,
```
("," is missing)

File: api/formatting.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      "startsOn" : "2017-10-16",
+      //...
+    },
+    {
+      "id" : 2
```
Comment (lucmir) [inline api/formatting.md]:
```
"id": 2,
```
("," is missing)

File: api/formatting.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+      "startsOn" : "2017-10-16",
+      //...
+    }
+  }
+}
+// do this instead
+{
+  "leave" : [
+    {
+      "id" : 1
```
Comment (lucmir) [inline api/formatting.md]:
```
"id": 1,
```
("," is missing)

File: api/formatting.md (hunk 4 above)
Comment (Stono) [reply api/formatting.md]: I'm confused, that'd make 1 a string, rather than a numeric - both are valid?

**Comment by MarcosX** [comment]
@lucmir @Stono I just updated the code with your suggestions, let me know what you think!

**Comment by juliofarah** [comment]
Hey all! 
Are we good to merge it?

**Pull request merged**